lib/repository.js

```diff
@@ -648,14 +649,20 @@ Repository.prototype.fetchAll = function(
   var repo = this;
 
   return repo.getRemotes().then(function(remotes) {
-    var fetchPromises = [];
+    var fetchPromise = Promise.resolve();
 
     remotes.forEach(function(remote) {
-      fetchPromises.push(
-        repo.fetch(remote, remoteCallbacks, autoTag, pruneAfter, callback));
+      fetchPromise.then(function() {
+        repo.fetch(remote, remoteCallbacks, autoTag, pruneAfter, callback);
```

Comment:
Touché. So it shouldn't have that passed in at all. Ahh I didn't even return the promise so the chain isn't working as I thought it would. I'll get this fixed up.